src/widgets/helper/ResizingTextEdit.cpp

```diff
@@ -257,25 +261,51 @@ void ResizingTextEdit::insertCompletion(const QString &completion)
 
 bool ResizingTextEdit::canInsertFromMimeData(const QMimeData *source) const
 {
-    if (source->hasImage())
+    if (source->hasImage() || source->hasFormat("text/plain"))
     {
-        return false;
+        return true;
     }
-    else if (source->hasFormat("text/plain"))
+    else
     {
-        return true;
+        return QTextEdit::canInsertFromMimeData(source);
     }
-    return false;
 }
 
 void ResizingTextEdit::insertFromMimeData(const QMimeData *source)
 {
-    if (!source->hasImage())
+    if (source->hasImage() || source->hasUrls())
+    {
+        this->imagePasted.invoke(source);
+    }
+    else
     {
         insertPlainText(source->text());
     }
 }
 
+void ResizingTextEdit::dragEnterEvent(QDragEnterEvent *event)
+{
+    if (event->mimeData()->hasImage() || event->mimeData()->hasUrls())
+    {
+        event->acceptProposedAction();
+    }
+    else
+    {
+        QTextEdit::dragEnterEvent(event);
+    }
+}
+
+void ResizingTextEdit::dropEvent(QDropEvent *event)
+{
+    if (event->mimeData()->hasImage() || event->mimeData()->hasUrls())
+    {
+        this->imagePasted.invoke(event->mimeData());
+    }
+    else  // allow for previous functionality of dropping text.
```

Comment:
This is a comment for the pr, not the code.

Reply:
Fixed in `a86367f4`